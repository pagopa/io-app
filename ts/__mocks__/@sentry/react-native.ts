export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
