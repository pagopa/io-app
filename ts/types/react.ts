import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
