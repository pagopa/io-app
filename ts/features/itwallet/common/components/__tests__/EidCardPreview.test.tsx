import { render } from "@testing-library/react-native";
import * as React from "react";
import { EidCardPreview } from "../EidCardPreview";

describe("EidCardPreview", () => {
  it("should match the snapshot", () => {
    const component = render(<EidCardPreview />);
    expect(component).toMatchSnapshot();
  });
});
