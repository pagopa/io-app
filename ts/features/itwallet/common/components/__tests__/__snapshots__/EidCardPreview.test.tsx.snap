// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EidCardPreview should match the snapshot 1`] = `
<View
  style={
    Object {
      "aspectRatio": 4.5,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "aspectRatio": 1.6,
        "borderRadius": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E8EBF1",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <SvgMock />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingTop": 12,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="bluegreyDark"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          style={
            Array [
              Object {
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          IDENTITÃ€ DIGITALE
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "borderLeftWidth": 9,
            "borderRadius": 8,
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          false,
          false,
        ]
      }
    />
  </View>
</View>
`;
